frontend/src/components/image/ImageSwipe.vue

```diff
@@ -0,0 +1,292 @@
+                            // http://photoswipe.com/documentation/faq.html#custom-pid-in-url
+                            for (let j = 0; j < items.length; j++) {
+                                if (items[j].pid == index) {
+                                    options.index = j;
+                                    break;
+                                }
+                            }
+                        } else {
+                            // in URL indexes start from 1
+                            options.index = parseInt(index, 10) - 1;
+                        }
+                    } else {
+                        options.index = parseInt(index, 10);
+                    }
+                    // exit if index not found
+                    if (isNaN(options.index)) {
+                        return;
+                    }
+                    if (disableAnimation) {
+                        options.showAnimationDuration = 0;
+                    }
+                    // Pass data to PhotoSwipe and initialize it
+                    gallery = new PhotoSwipe(pswpElement, PhotoSwipeUI_Default, items, Object.assign(options, that.options));
+
+                    gallery.listen('gettingData', function(index, item) {
+                        if (item.w < 1 || item.h < 1) { // unknown size
+                            var img = new Image();
+                            img.onload = function() { // will get size after load
+                                item.w = this.width; // set image width
+                                item.h = this.height; // set image height
+                                gallery.invalidateCurrItems(); // reinit Items
+                                gallery.updateSize(true); // reinit Items
+                            };
+                            img.src = item.src; // let's download image
+                        }
+                    });
+
+                    gallery.init();
+                };
+                // loop through all gallery elements and bind events
+                let galleryElements = document.querySelectorAll(gallerySelector);
+                for (let i = 0, l = galleryElements.length; i < l; i++) {
+                    galleryElements[i].setAttribute('data-pswp-uid', i + 1);
+                    galleryElements[i].onclick = onThumbnailsClick;
+                }
+                // Parse URL and open gallery if it contains #&pid=3&gid=1
+                let hashData = photoswipeParseHash();
+                if (hashData.pid && hashData.gid) {
+                    openPhotoSwipe(hashData.pid, galleryElements[hashData.gid - 1], true, true);
+                }
+            };
+            initPhotoSwipeFromDOM('.my-gallery');
+        },
+        methods: {
+            rotate: function(newAngle) {
+                this.angle = this.angle + newAngle
+                this.$el.querySelectorAll('.pswp__img').forEach(i => i.style.transform = `rotate(${this.angle}deg)`)
```

Comment:
First of all, you can't use operation with DOM (querySelectorAll, getElementById etc.). Second we don't need to rotate image